testdata/src/go.uber.org/contracts/userdefinedfunctions.go

```diff
@@ -0,0 +1,733 @@
+//  Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package contracts: This file tests the contract of "ok" form for user defined and standard library functions.
+//
+// <nilaway no inference>
+package contracts
```

Comment:
Appreciate the comprehensive test suite. As mentioned in other comments, perhaps we can add some cases that devs shadow the builtin types, as well as defining a constant for `true` or `false` :)

Reply:
Done for constants. Shadowing built-in types is not so straightforward, so will do it subsequently.